repository/kafka/src/main/java/com/expediagroup/streamplatform/streamregistry/repository/kafka/DefaultRepository.java

```diff
@@ -48,9 +48,19 @@ abstract class DefaultRepository<
   private final Converter<ME, MK, SK, SS> converter;
   private final Class<SK> stateKeyClass;
 
-  //TODO deprecate - split specification from status so they are saveable separately.
+  /**
+   * @deprecated Use {link {@link #saveSpecification(com.expediagroup.streamplatform.streamregistry.model.Entity)}} or
+   * {link {@link #saveStatus(com.expediagroup.streamplatform.streamregistry.model.Entity)}}
+   * to save the entity specification or entity status independently.
+   */
+  @Deprecated
   @Override
   public ME save(ME entity) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public ME saveSpecification(ME entity) {
     Optional<ME> existing = findById(entity.getKey());
     Entity<SK, SS> stateEntity = converter.convertEntity(entity);
     List<CompletableFuture<Void>> futures = new ArrayList<>();
```

Comment:
Do we need these lists of futures now? Could we do similar to what we do in the agents...

```java
  private void sendSync(Event<SK, SS> event) {
    sender.send(event).join();
  }
```

Usages then become:
```java
sendSync(Event.specification(stateEntity.getKey(), stateEntity.getSpecification()));
```

Reply:
yep, missed that. done